api-js/src/loaders/domains/load-domain-connections-by-user-id.js

```diff
@@ -0,0 +1,164 @@
+const { aql } = require('arangojs')
+const { fromGlobalId, toGlobalId } = require('graphql-relay')
+
+const domainLoaderConnectionsByUserId = (
+  query,
+  userId,
+  cleanseInput,
+) => async ({ after, before, first, last }) => {
+  let afterTemplate = aql``
+  let beforeTemplate = aql``
+
+  const userDBId = `users/${userId}`
+
+  if (typeof after !== 'undefined') {
+    const { id: afterId } = fromGlobalId(cleanseInput(after))
+    afterTemplate = aql`FILTER TO_NUMBER(domain._key) > TO_NUMBER(${afterId})`
+  }
+
+  if (typeof before !== 'undefined') {
+    const { id: beforeId } = fromGlobalId(cleanseInput(before))
+    beforeTemplate = aql`FILTER TO_NUMBER(domain._key) < TO_NUMBER(${beforeId})`
+  }
+
+  let limitTemplate = aql``
+  if (typeof first !== 'undefined' && typeof last === 'undefined') {
+    limitTemplate = aql`SORT domain._key ASC LIMIT TO_NUMBER(${first + 1})`
+  } else if (typeof first === 'undefined' && typeof last !== 'undefined') {
+    limitTemplate = aql`SORT domain._key DESC LIMIT TO_NUMBER(${last + 1})`
+  } else if (typeof first !== 'undefined' && typeof last !== 'undefined') {
+    console.warn(
+      `User: ${userId} tried to have first and last set in domain connection query`,
+    )
+    throw new Error(
+      'Error, unable to have first, and last set at the same time.',
+    )
+  }
+
+  let affiliatedOrgsCursor
+  try {
+    affiliatedOrgsCursor = await query`
+      LET userAffiliations = (FOR v, e IN 1..1 ANY ${userDBId} affiliations RETURN e._from)
+      RETURN userAffiliations
+    `
+  } catch (err) {
+    console.error(
+      `Database error occurred while user: ${userId} was trying to query affiliated organizations in loadDomainsByUser.`,
+    )
+    throw new Error('Unable to query organizations. Please try again.')
+  }
+
+  let affiliatedOrgs
+  try {
+    affiliatedOrgs = await affiliatedOrgsCursor.next()
+  } catch (err) {
+    console.error(
+      `Cursor error occurred while user: ${userId} was trying to gather affiliated organizations in loadDomainsByUser.`,
+    )
+    throw new Error('Unable to load organizations. Please try again.')
+  }
+
+  let claimedDomainsCursor, claimedDomains, orgId
+  const orgDomains = []
+
+  for (let i = 0; i < affiliatedOrgs.length; i++) {
```

Comment:
Look at the above comment, this way we only ever have to send one query, so it won't be prone to failing as often when looping through Id's